<launch>
    <param name="use_sim_time" value="false" />

    <arg name="bag_path" default="/home/abs8rng/BagFiles/Weedmobile/2015/12/low_cost_navigation/Trimble_IMU_Piksi_Asphalt_1/" />
    <arg name="bag_name" default="Trimble_IMU_Piksi_Asphalt_1_2015-12-10-15-39-44_0" />
    <arg name="Rate" default="1" />
    <arg name="loop" default="" />
    <arg name="Topic_name" default="/sensor/gps/trimble/fix" />

    <arg name="plot_gps" default="false" />
    <arg name="imu_gui" default="false" />
    <arg name="plot_imu" default="false" />
    <arg name="plot_odom" default="false" />

    <node pkg="rosbag" type="play" name="player" output="screen" args="$(arg loop) -r $(arg Rate) $(arg bag_path)$(arg bag_name).bag" />

    <node pkg="rqt_robot_monitor" type="rqt_robot_monitor" name="rqt_robot_monitor" />

<!-- Plot_gps -->
<group if="$(arg plot_gps)">
    <node pkg="test_code" type="plot_piksi_vs_trimble_orientation.py" name="plot_piksi_vs_trimble_orientation" output="screen" />
    <node pkg="test_code" type="plot_piksi_vs_trimble.py" name="plot_piksi_vs_trimble" output="screen"/>
</group>

<!-- imu_gui -->
<group if="$(arg imu_gui)">
  <node pkg="imu_visualization" type="display_3D_visualization.py" name="um7_3D_visualization_node" output="screen">
	<param name="imu_topic" value="/sensor/imu/um7/data" />
	<param name="imu_name" value="CHRobotics_um7" />
	<param name="mag_topic" value="/sensor/imu/um7/magfield_msg" />
  </node>
  <node pkg="imu_visualization" type="display_3D_visualization.py" name="Xsens_3D_visualization_node" output="screen">
	<param name="imu_topic" value="/sensor/imu/xsens_mti/data" />
	<param name="imu_name" value="xsens_mti" />
	<param name="mag_topic" value="/sensor/imu/xsens_mti/mag_calib" />
  </node>

<!--
  <node pkg="imu_visualization" type="display_3D_visualization.py" name="um7_3D_visualization_node" output="screen">
	<param name="imu_topic" value="imu/data" />
	<param name="imu_name" value="CHRobotics_um7" />
	<param name="mag_topic" value="/imu/mag_msg" />
  </node> -->
</group>


	<!-- <node pkg="rqt_plot" type="rqt_plot" name="rqt_bag_file_plot" output="screen" args="$(arg Topic_name)" /> -->

</launch>
